core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -457,23 +457,28 @@ class StatementAnalyzer
 
     public Scope analyze(Node node)
     {
-        return analyze(node, Optional.empty());
+        return analyze(node, Optional.empty(), true);
     }
 
     public Scope analyze(Node node, Scope outerQueryScope)
```

Comment:
Consider getting rid of this method and making the caller explicitly say top level or not.

Reply:
Actually, this method implies it's not top-level (since there's an outer query scope). It's the method below that is not necessary. I'll update its usages and remove it.